core/src/main/java/io/dekorate/kubernetes/decorator/AddDockerConfigJsonSecretDecorator.java

```diff
@@ -68,16 +68,21 @@ public AddDockerConfigJsonSecretDecorator(String name, Path existingConfigJson,
   @Override
   public void visit(KubernetesListBuilder list) {
     String name = Strings.isNotNullOrEmpty(this.name) ? this.name : getMandatoryDeploymentMetadata(list, ANY).getName();
-    Secret secret = new SecretBuilder()
+    SecretBuilder secret = new SecretBuilder()
         .withNewMetadata()
         .withName(name)
         .withAnnotations(this.annotations)
         .endMetadata()
         .withType(TYPE)
-        .addToData(DOT_DOCKER_CONFIG_JSON, this.content)
-        .build();
+        .addToData(DOT_DOCKER_CONFIG_JSON, this.content);
```

Comment:
Why don't you just make that key configuralbe? It will save you the `doUpate` and extending thing.
Alterntatively, you can just create a decorator that applies the name of they key.

Reply:
Because we need these two keys. The `config.json` is for the step and the `.dockerconfigjson` one is the standard for Kubernetes image pull secret. 
I wrote a commend in the class that explained why it was required.